server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -1017,6 +1018,9 @@ public IndexShard createShard(
                 .setSource(mapping.source().string(), XContentType.JSON)
                 .get();
         }, this);
+        if (indexService.getIndexSettings().isSegRepEnabled() && shardRouting.primary() == false) {
+            indexShard.setSegmentReplicationState(new SegmentReplicationState(indexShard.routingEntry(), sourceNode, targetNode));
```

Comment:
We shouldn't need to set this object during shard creation, only when the first segrep event starts.

Reply:
Sure, I will remove this invokation